docs/markdown/Helm/helm-deployments.md

```diff
@@ -0,0 +1,197 @@
+```
+
+This should work any kind of Kubernetes resource that will result in Docker image being deployed into Kubernetes, like `Deployment`, `StatefulSet`, `ReplicaSet`, `CronJob`, etc. Please get in touch with us in case you find Pants was not capable to infer dependencies in any of your `helm_deployment` targets by either [opening a GitHub issue](https://github.com/pantsbuild/pants/issues/new/choose) or [joining our Slack](doc:getting-help).
+
+> 📘 How the Docker image reference is calculated during deployment?
+> 
+> Pants' will rely on the behaviour of the `docker_image` target when it comes down to generate the final image reference. Since a given image may have more than one valid image reference, **Pants will try to use the first one that is not tagged as `latest`**, falling back into `latest` if none could be found.
+> It's good practice to publish your Docker images using tags other than `latest` and Pants preferred behaviour is to choose those are this guarantees that the _version_ of the Docker image being deployed is the expected one.
+
+Value override files
+--------------------
+
+As seen in the initial examples of the usage of the `helm_deployment` target, it can be seen that some of the source value files used the word `override` in the file name.
+
+When using deployments that may have more than one YAML file as the source of configuration values, it is recommended use the `override` word in those filenames that are meant to override values coming from other more common or general sources.
+
+Besides, you may be interested in organizing your deployment source files using a nested folder structure as the following:
+
+```
+src/deployment/config_maps.yaml
+src/deployment/services.yaml
+src/deployment/dev/services.yaml
+src/deployment/uat/services.yaml
+```
+
+```text src/deployment/BUILD
+helm_deployment(name="dev", dependencies=["//src/chart"], sources=["*.yaml", "dev/*.yaml"])
+
+helm_deployment(name="uat", dependencies=["//src/chart"], sources=["*.yaml", "uat/*.yaml"])
+```
+
+In this case, files that are in nested folders will also act as value override files.
+
+We believe that this approach gives enough flexibility to organise your deployment sources as you better prefer while providing an intuitive way to Pants' users to ensure that the list of value file sources is in the right order every single time.
+
+In addition to value files, you can also use inline values in your `helm_deployment` targets by means of the `values` field. All inlines values that are set this way will override any entry that may come from value files, whether the source file was an override file or not. The relationship between the three scopes can be imagined as follows:
+
+```
+Value files < Value override files < Inline values
+```
+
+Deploying
+---------
+
+Continuing with the example in the previous section, we can deploy it into Kubernetes using the command `./pants experimental-deploy src/deployment`. This will go trigger the following steps:
```

Comment:
```suggestion
Continuing with the example in the previous section, we can deploy it into Kubernetes using the command `./pants experimental-deploy src/deployment`. This will trigger the following steps:
```